src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoader.cs

```diff
@@ -63,14 +63,30 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                 // This allows us to correctly handle retargetable assemblies, redirects, etc.
                 if (result == null)
                 {
-                    string assemblyName = ((AppDomain)sender).ApplyPolicy(args.Name);
+                    var appDomain = (AppDomain)sender;
+                    string assemblyName = appDomain.ApplyPolicy(args.Name);
 
                     // If after applying the current policy, we now have a different target assembly name, attempt to load that 
                     // assembly
                     if (string.Compare(assemblyName, args.Name) != 0)
                     {
                         result = Assembly.Load(assemblyName);
                     }
+
+
+                    if (context == null && result == null)
+                    {
+                        // Fallback. 
+                        var shortName = new AssemblyName(assemblyName).Name;
```

Comment:
I was chatting with @MikeStall and proposed this to unblock him, but not as a permanent solution. We need to properly address the issue as we definitely don't want to have this behavior enabled as it would lead to a lot of issues (including leaking any runtime assemblies, increasing the risk of breaking changes and worse, causing bindings to incorrect assembly versions)